scripts/modules/task-manager/scope-adjustment.js

```diff
@@ -0,0 +1,828 @@
+/**
+ * scope-adjustment.js
+ * Core logic for dynamic task complexity adjustment (scope-up and scope-down)
+ */
+
+import { z } from 'zod';
+import {
+	log,
+	readJSON,
+	writeJSON,
+	getCurrentTag,
+	readComplexityReport,
+	findTaskInComplexityReport
+} from '../utils.js';
+import {
+	generateObjectService,
+	generateTextService
+} from '../ai-services-unified.js';
+import { findTaskById, taskExists } from '../task-manager.js';
+import analyzeTaskComplexity from './analyze-task-complexity.js';
+import { findComplexityReportPath } from '../../../src/utils/path-utils.js';
+
+/**
+ * Valid strength levels for scope adjustments
+ */
+const VALID_STRENGTHS = ['light', 'regular', 'heavy'];
+
+/**
+ * Statuses that should be preserved during subtask regeneration
+ * These represent work that has been started or intentionally set by the user
+ */
+const PRESERVE_STATUSES = [
+	'done',
+	'in-progress',
+	'review',
+	'cancelled',
+	'deferred',
+	'blocked'
+];
```

Comment:
I believe we have the statuses defined somewhere in src/constants, use those instead.